dataland-frontend/src/components/resources/companyCockpit/FrameworkSummaryPanel.vue

```diff
@@ -77,16 +78,23 @@ const injectedUseMobileView = inject<{ value: boolean }>("useMobileView");
 const useMobileView = computed<boolean | undefined>(() => injectedUseMobileView?.value);
 
 const getKeycloakPromise = inject<() => Promise<Keycloak>>("getKeycloakPromise");
-const isUserUploader = ref<boolean>();
+const isUserAllowedToUpload = ref<boolean>();
 onBeforeMount(() => {
   checkIfUserHasRole(KEYCLOAK_ROLE_UPLOADER, getKeycloakPromise)
     .then((result) => {
-      isUserUploader.value = result;
+      isUserAllowedToUpload.value = result;
+    })
+    .catch((error) => console.log(error));
+  isUserDataOwnerForCompany(props.companyId, getKeycloakPromise)
+    .then((result) => {
+      if (!isUserAllowedToUpload.value) {
+        isUserAllowedToUpload.value = result;
+      }
```

Comment:
Du könntest hier einen API-call sparen, indem du Folgendes tust:

- Check erst ob der user die Rolle "Uploader" hat
- Wenn ja, dann setzt `isUserAllowedToUpload` auf `true` und tu nichts weiter (=> Api-Call gespart)
- Wenn nein, dann - und nur dann - check ob der user vllt data-owner ist

So wird nur nach der data-ownership geschaut, wenn der user nicht bereits uploader-Rechte hat.

Reply:
Finde ich gut die Idee, ist jetzt umgesetzt

Reply:
Die Umsetzung ist zwar richtig, aber man sollte so eine "pyramid of doom" grundsätzlich vermeiden.
Siehe (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises) 

Promises sollten für Leserlichkeit so gechaint werden:

![image](https://github.com/d-fine/Dataland/assets/99255296/fa3dcdf7-c44d-4b7b-9da6-0be15435c03a)


Ich denke das sollte hier möglich sein.